lib/plugins/aws/package/compile/events/websockets/lib/api.js

```diff
@@ -14,7 +14,7 @@ module.exports = {
     this.websocketsApiLogicalId = this.provider.naming.getWebsocketsApiLogicalId();
 
     const RouteSelectionExpression =
-      this.serverless.service.provider.websocketsApiRouteSelectionExpression ||
```

Comment:
We should still support the old syntax, but checking the new one first when resolving the property

Reply:
What is currently causing you troubles? From what I see in the errors - it's failing because it tries to access `websocket.apiName` when `websocket` is not defined at all - we have to keep that in mind that people might not define anything under `provider.websocket` so it can be undefined